src/main/Application.h

```diff
@@ -157,6 +181,33 @@ class Application
         APP_NUM_STATE
     };
 
+    enum class AppMode
+    {
+        // Application will connect to networks (unless Cfg.RUN_STANDALONE),
+        // bind to a database and bucket list, and receive and apply
+        // consensus transactions. "Normal operation".
+        RUN_LIVE_NODE,
+
+        // Application will connect to networks but _not_ establish a database
+        // of any sort, nor participate in consensus, apply transactions or
+        // even verify them. Simply relays traffic between nodes.
+        RELAY_LIVE_TRAFFIC,
+
+        // Application will not connect to networks nor establish a database
+        // of any sort. Simply reads history archive content and replays
+        // transactions against _in memory_ ledger, while writing metadata
+        // stream.
+        REPLAY_HISTORY_FOR_METADATA
+    };
+
+    // AppModes should be accessed using either a narrow centralized query like
+    // one of the following static functions, or an exhaustive switch on the
+    // enum. These options give the least chance of missing or combining cases
+    // incorrectly.
+    static bool modeHasOverlay(AppMode m);
+    static bool modeHasDatabase(AppMode m);
+    static bool modePublishesHistory(AppMode m);
```

Comment:
I expect we'll need a few more such mode judgements but I didn't have a clear enough idea of which ones we'd need. I'm going to leave this as-is for now since this PR isn't switching bucket behaviour (yet).